optuna/visualization/matplotlib/_pareto_front.py

```diff
@@ -74,13 +81,54 @@ def objective(trial):
     """
 
     _imports.check()
-
-    if len(study.directions) == 2:
-        return _get_pareto_front_2d(study, target_names, include_dominated_trials, axis_order)
-    elif len(study.directions) == 3:
-        return _get_pareto_front_3d(study, target_names, include_dominated_trials, axis_order)
+    if axis_order:
+        warnings.warn(
+            "``axis_order`` option is deprecated. This feature will be removed in the future.",
+            DeprecationWarning,
+        )
+    trials = _get_trials(study, include_dominated_trials)
+    if targets is not None and axis_order is not None:
+        raise ValueError(
+            "Using both ``targets`` and ``axis_order`` is not supported. "
+            " Use either `targets` or `axis_order`."
+        )
+    if targets is None:
+        if len(study.directions) == 2:
+            targets = _targets_default_2d
+        elif len(study.directions) == 3:
+            targets = _targets_default_3d
+        else:
+            raise ValueError(
+                "``plot_pareto_front`` function only supports 2 or 3 objective"
+                " studies when using ``targets`` is ``None``. Please use`targets`"
+            )
+    target_values = [targets(t) for t in trials]
+    n_targets = len(study.directions)
+    if target_names is not None:
+        n_targets = len(target_names)
+    if len(target_values) > 0:
+        n_targets = len(target_values[0])
```

Comment:
If `targets` is specified, `len(targeta_values) == 0` (`len(trials) = 0`) and `target_names` is not specified, then we cannot decide the `n_targets` in principle. I mean the actual `n_targets` is not necessarily same as `len(study.directions)` in this situation. How about the following?

```
if len(target_values) > 0:
    n_targets = len(target_values[0])
elif target_names is not None:
    n_targets = len(target_names)
elif targets is None:
    n_targets = len(study.directions)
else:
    raise ValueError("For empty studies with the number of objectives greater than 3, ``target_names`` must be specified.")
```

Reply:
Does the current test cover the all of above cases? (Sorry I'm not sure about it.)

Reply:
Thank you for the suggestion.
I think the last elif statement is not required and else statement should be changed in your situation.
How about the following?
```
    if len(target_values) > 0:
        n_targets = len(target_values[0])
    elif target_names is not None:
        n_targets = len(target_names)
    else:
        n_targets = len(study.directions)
```

Also, in this next if statement, it is checked whether `n_targets` is 2 and 3.